recipes-connectivity/prplmesh/prplmesh/004-get_hostapd_ctrl_iface_pathes_from_conf.patch

```diff
@@ -0,0 +1,286 @@
+ set(BEEROCKS_LOG_STDOUT_ENABLED     "false")
+ set(BEEROCKS_LOG_SYSLOG_ENABLED     "false")
+ 
++# hostapd ctrl ifaces paths
++if(NOT DEFINED BEEROCKS_HOSTAP_WLAN1_CTRL_IFACE)
++    set(BEEROCKS_HOSTAP_WLAN1_CTRL_IFACE "/var/run/hostapd/${BEEROCKS_WLAN1_IFACE}")
++endif()
++if(NOT DEFINED BEEROCKS_HOSTAP_WLAN2_CTRL_IFACE)
++    set(BEEROCKS_HOSTAP_WLAN2_CTRL_IFACE "/var/run/hostapd/${BEEROCKS_WLAN2_IFACE}")
++endif()
++if(NOT DEFINED BEEROCKS_HOSTAP_WLAN3_CTRL_IFACE)
++    set(BEEROCKS_HOSTAP_WLAN3_CTRL_IFACE "/var/run/hostapd/${BEEROCKS_WLAN3_IFACE}")
++endif()
++
++# wpa supplicant ctrl ifaces paths
++if(NOT DEFINED BEEROCKS_WPA_SUPPLICANT_WLAN1_CTRL_IFACE)
++    set(BEEROCKS_WPA_SUPPLICANT_WLAN1_CTRL_IFACE "/var/run/wpa_supplicant/${BEEROCKS_WLAN1_IFACE}")
++endif()
++if(NOT DEFINED BEEROCKS_WPA_SUPPLICANT_WLAN2_CTRL_IFACE)
++    set(BEEROCKS_WPA_SUPPLICANT_WLAN2_CTRL_IFACE "/var/run/wpa_supplicant/${BEEROCKS_WLAN2_IFACE}")
++endif()
++if(NOT DEFINED BEEROCKS_WPA_SUPPLICANT_WLAN3_CTRL_IFACE)
++    set(BEEROCKS_WPA_SUPPLICANT_WLAN3_CTRL_IFACE "/var/run/wpa_supplicant/${BEEROCKS_WLAN3_IFACE}")
++endif()
++
+ # Platform specific flags
+ if (TARGET_PLATFORM STREQUAL "openwrt")
+     if (TARGET_PLATFORM_TYPE STREQUAL "ugw")
+diff --git a/agent/config/beerocks_agent.conf.in b/agent/config/beerocks_agent.conf.in
+index 0b6ddae9..c1f2a07e 100644
+--- a/agent/config/beerocks_agent.conf.in
++++ b/agent/config/beerocks_agent.conf.in
+@@ -28,18 +28,24 @@ radio_identifier=00:00:00:00:00:00
+ hostap_iface_type=WIFI_INTEL
+ hostap_ant_gain=0
+ enable_repeater_mode=@BEEROCKS_REPEATER_MODE@
++hostap_ctrl_iface=@BEEROCKS_HOSTAP_WLAN1_CTRL_IFACE@
++wpa_supplicant_ctrl_iface=@BEEROCKS_WPA_SUPPLICANT_WLAN1_CTRL_IFACE@
+ 
+ [agent1]
+ radio_identifier=00:00:00:00:00:02
+ hostap_iface_type=WIFI_INTEL
+ hostap_ant_gain=0
+ enable_repeater_mode=0
++hostap_ctrl_iface=@BEEROCKS_HOSTAP_WLAN2_CTRL_IFACE@
++wpa_supplicant_ctrl_iface=@BEEROCKS_WPA_SUPPLICANT_WLAN2_CTRL_IFACE@
+ 
+ [agent2]
+ radio_identifier=00:00:00:00:00:02
+ hostap_iface_type=WIFI_INTEL
+ hostap_ant_gain=0
+ enable_repeater_mode=0
++hostap_ctrl_iface=@BEEROCKS_HOSTAP_WLAN3_CTRL_IFACE@
++wpa_supplicant_ctrl_iface=@BEEROCKS_WPA_SUPPLICANT_WLAN3_CTRL_IFACE@
+ 
+ [log]
+ log_global_levels=error,info,warning,fatal,trace,debug
+diff --git a/agent/src/beerocks/slave/ap_manager_thread.cpp b/agent/src/beerocks/slave/ap_manager_thread.cpp
+index 4ac24d50..92ec656d 100644
+--- a/agent/src/beerocks/slave/ap_manager_thread.cpp
++++ b/agent/src/beerocks/slave/ap_manager_thread.cpp
+@@ -112,6 +112,7 @@ void ap_manager_thread::ap_manager_config(ap_manager_conf_t &conf)
+ 
+     bwl::hal_conf_t hal_conf;
+     hal_conf.ap_acs_enabled = acs_enabled;
++    hal_conf.wpa_ctrl_path = conf.wpa_ctrl_path;
+ 
+     // Create a new AP HAL instance
+     ap_wlan_hal = bwl::ap_wlan_hal_create(
+diff --git a/agent/src/beerocks/slave/ap_manager_thread.h b/agent/src/beerocks/slave/ap_manager_thread.h
```

Comment:
It is a bad idea do patch on the source code. Can you please move changes for source code to the main prplmesh project?

Reply:
There are prplmesh PRs for each of patch presented in `meta-prplmesh`
See links in this PR description